### ISSUE TITLE
vet and vetshadow have spurious output

### ISSUE THREAD

**Issue by joefitzgerald** [alecthomas/gometalinter#179]
Actual:

`cmd/goulash/vet: main.go: main.go:17::error: 9: expected ';', found '(' (vet)`

Expected:

`cmd/goulash/vet: main.go:17::error: 9: expected ';', found '(' (vet)`

**Comment by joefitzgerald** [comment]
Related: https://github.com/joefitzgerald/gometalinter-linter/issues/60

**Comment by alecthomas** [comment]
This is a bug/behavior of vet. Parse errors are reported differently to actual vet errors.

**Comment by alecthomas** [comment]
This has always been present. Example:

Parse error:

```
[alec@cavern:~/foo]cat t.go
package main

type foo strct {
	a string `a b c`
}

func main() {}
[alec@cavern:~/foo]go tool vet .
vet: t.go: t.go:3:16: expected ';', found '{'
```

Vet error:

```
[alec@cavern:~/foo]cat t.go
package main

type foo struct {
	a string `a b c`
}

func main() {}
[alec@cavern:~/foo]go tool vet .
t.go:4: struct field tag `a b c` not compatible with reflect.StructTag.Get: bad syntax for struct tag pair
```
